src/main/scala/is/hail/expr/Type.scala

```diff
@@ -181,6 +181,18 @@ sealed abstract class Type extends Serializable { self =>
 
   val partitionKey: Type = this
 
+  def typedOrderedKey[PK, K] = new OrderedKey[PK, K] {
```

Comment:
Shouldn't this require `[PK, K <: PK]`?

Reply:
No.  In general PK is a subset of K, like Locus and Variant.

Reply:
`K` must be a subtype (`<:`) of `PK`, otherwise the next line

```scala
def project(key: K): PK = key.asInstanceOf[PK]
```

will certainly fail, right?